openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,218 @@ def maintained_selection():
+
+    # Use displayLights setting from instance
+    key = "displayLights"
+    preset["viewport_options"][key] = instance.data[key]
+
+    # Override transparency if requested.
+    transparency = instance.data.get("transparency", 0)
+    if transparency != 0:
+        preset["viewport2_options"]["transparencyAlgorithm"] = transparency
+
+    # Isolate view is requested by having objects in the set besides a
+    # camera. If there is only 1 member it'll be the camera because we
+    # validate to have 1 camera only.
+    if instance.data["isolate"] and len(instance.data["setMembers"]) > 1:
+        preset["isolate"] = instance.data["setMembers"]
+
+    # Show/Hide image planes on request.
+    image_plane = instance.data.get("imagePlane", True)
+    if "viewport_options" in preset:
+        preset["viewport_options"]["imagePlane"] = image_plane
+    else:
+        preset["viewport_options"] = {"imagePlane": image_plane}
+
+    # Force viewer to False in call to capture because we have our own
+    # viewer opening call to allow a signal to trigger between
+    # playblast and viewer
+    preset["viewer"] = False
+
+    # Update preset with current panel setting
+    # if override_viewport_options is turned off
+    override_viewport_options = (
+        capture_preset["Viewport Options"]["override_viewport_options"]
+    )
+    if not override_viewport_options:
+        panel_preset = capture.parse_view(instance.data["panel"])
+        panel_preset.pop("camera")
+        preset.update(panel_preset)
+
+    return preset
+
+
+@contextlib.contextmanager
+def viewport_default_options(preset, instance):
+    # Disable Pan/Zoom.
+    pan_zoom = cmds.getAttr("{}.panZoomEnabled".format(preset["camera"]))
+    preset.pop("pan_zoom", None)
+    preset["camera_options"]["panZoomEnabled"] = instance.data["panZoom"]
+
+    viewport_defaults = {}
+    # Need to explicitly enable some viewport changes so the viewport is
+    # refreshed ahead of playblasting.
+    try:
+        keys = [
+            "useDefaultMaterial",
+            "wireframeOnShaded",
+            "xray",
+            "jointXray",
+            "backfaceCulling",
+            "textures"
+        ]
+        for key in keys:
+            viewport_defaults[key] = cmds.modelEditor(
+                instance.data["panel"], query=True, **{key: True}
+            )
+            if preset["viewport_options"][key]:
+                cmds.modelEditor(
+                    instance.data["panel"], edit=True, **{key: True}
+                )
+        yield
+    finally:
+        # Restoring viewport options.
+        if viewport_defaults:
+            cmds.modelEditor(
+                instance.data["panel"], edit=True, **viewport_defaults
+            )
+        try:
+            cmds.setAttr(
+                "{}.panZoomEnabled".format(preset["camera"]), pan_zoom)
+        except RuntimeError:
+            self.log.warning("Cannot restore Pan/Zoom settings.")
```

Comment:
```suggestion
def viewport_default_options(preset):
    """Context manager used by `render_capture_preset`.
        
    We need to explicitly enable some viewport changes so the viewport is
    refreshed ahead of playblasting.
    
    """
    # TODO: Clarify in the docstring WHY we need to set it ahead of
    #  playblasting. What issues does it solve?
    viewport_defaults = {}
    try:
        keys = [
            "useDefaultMaterial",
            "wireframeOnShaded",
            "xray",
            "jointXray",
            "backfaceCulling",
            "textures"
        ]
        for key in keys:
            viewport_defaults[key] = cmds.modelEditor(
                preset["panel"], query=True, **{key: True}
            )
            if preset["viewport_options"][key]:
                cmds.modelEditor(
                    preset["panel"], edit=True, **{key: True}
                )
        yield
    finally:
        # Restoring viewport options.
        if viewport_defaults:
            cmds.modelEditor(
                preset["panel"], edit=True, **viewport_defaults
            )
```

Bonus points for adding a docstring that actually explains **WHY** this is needed. Currently it's unclear why this is needed. The comment:
```
    # Need to explicitly enable some viewport changes so the viewport is
    # refreshed ahead of playblasting.
```
Doesn't tell me much either. :( Why do we need to explicitly enable it?

Reply:
Fixed in #6081 but added to do about the comment since I'm not sure why it needs to be done.